assembl/static2/js/app/components/common/formControlWithLabel.jsx

```diff
@@ -7,25 +8,49 @@
 import React from 'react';
 import { ControlLabel, FormGroup, FormControl, HelpBlock } from 'react-bootstrap';
 import { I18n } from 'react-redux-i18n';
+import { type rawContentState } from 'draft-js';
 
 import RichTextEditor from './richTextEditor';
 import { getValidationState } from '../administration/voteSession/voteProposalForm';
 
-class FormControlWithLabel extends React.Component {
-  constructor(props) {
+type FormControlWithLabelProps = {
+  value: string | rawContentState,
+  required: boolean,
+  onChange: Function,
+  type: string,
+  disabled: boolean,
+  label: string,
+  labelAlwaysVisible: boolean,
+  componentClass: string,
+  formControlProps: Object,
+  id: string,
+  validationErrors?: Array<ErrorDef>
+};
+
+type FormControlWithLabelState = {
+  errorMessage?: string,
+  validationErrors?: Array<ErrorDef>,
+  validationState: ?string
+};
+
+class FormControlWithLabel extends React.Component<*, FormControlWithLabelProps, FormControlWithLabelState> {
+  props: FormControlWithLabelProps;
+
+  state: FormControlWithLabelState;
+
+  constructor(props: FormControlWithLabelProps) {
     super(props);
     this.state = this.getStateFromProps(props);
-    this.setValidationState = this.setValidationState.bind(this);
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentWillReceiveProps(nextProps: FormControlWithLabelProps) {
     if (nextProps.validationErrors !== this.props.validationErrors) {
       this.setState(this.getStateFromProps(nextProps));
     }
   }
 
-  getStateFromProps = (props) => {
-    const errors = props.validationErrors;
+  getStateFromProps = ({ validationErrors }: FormControlWithLabelProps) => {
+    const errors = validationErrors && validationErrors;
```

Comment:
why ?! It means that errors may be `false`

Reply:
Not false but falsy, it can be undefined or null I believe. There is a flow error when you don't do this. I think they have a github issue regarding that

Reply:
ok so you should at least add a comment (and probably a link to the github issue). Can you tell me what is the flow error if you remove it please?

Reply:
My bad, it is not the case here. And this const errors = validationErrors is actually useless, I'll remove it